src/main/java/com/smartsheet/wiremock/extensions/ApiScenarioTransformer.java

```diff
@@ -248,17 +244,69 @@ private static String diffExpectedHeaders(Request request, JsonNode scenarioHead
 
 		private static String diffExpectedHeader(Request request, Map.Entry<String, JsonNode> header) {
 			if (!request.containsHeader(header.getKey())) {
-				return String.format("headers: Expected %s, but not found\n", header.getKey());
+				return String.format("Headers: Expected %s, but not found. ", header.getKey());
 			}
 
 			String headerValue = header.getValue().asText();
 			String requestValue = request.getHeader(header.getKey());
 
 			if (!headerValue.equals(requestValue)) {
-				return formatAssert("headers - " + header.getKey(), headerValue, requestValue);
+				return formatAssert("Header - " + header.getKey(), headerValue, requestValue);
 			}
 
 			return "";
 		}
+
+		private static String diffUnexpectedHeaders(Request request, JsonNode scenarioHeaders) {
+			StringBuilder unexpectedHeaderDiff = new StringBuilder();
+			for (HttpHeader header : request.getHeaders().all()) {
+				if (!scenarioHeaders.has(header.key())) {
+					unexpectedHeaderDiff.append(String.format("Headers: Contains %s, but not expected. ", header.key()));
+				}
+			}
+
+			return unexpectedHeaderDiff.toString();
+		}
+
+		private static String diffExpectedQueryParams(Request request, JsonNode scenarioQueryParams) {
+			Iterator<Map.Entry<String, JsonNode>> scenarioQueryParamIterator = scenarioQueryParams.fields();
+
+			StringBuilder queryParamDiff = new StringBuilder();
+			while(scenarioQueryParamIterator.hasNext()) {
+				Map.Entry<String, JsonNode> queryParam = scenarioQueryParamIterator.next();
+
+				queryParamDiff.append(diffExpectedQueryParam(request, queryParam));
+			}
+
+			return queryParamDiff.toString();
+		}
+
+		private static String diffExpectedQueryParam(Request request, Map.Entry<String, JsonNode> queryParam) {
+		    QueryParameter requestParam = request.queryParameter(queryParam.getKey());
+		    if(requestParam == null || requestParam.key() == null || requestParam.key().isEmpty()){
+				return String.format("Query Parameters: Expected %s, but not found. ", queryParam.getKey());
+			}
+
+			if (!requestParam.containsValue(queryParam .getValue().asText())) {
```

Comment:
The indentation in this file ended up super funky. Could you run an auto-format command on it?